ansible/roles/test/files/ptftests/arista.py

```diff
@@ -89,10 +95,35 @@ def do_cmd(self, cmd, prompt = None):
 
         if cmd is not None:
             self.shell.send(cmd + '\n')
+            # wait 0.2s for the executing
+            time.sleep(0.2)
 
         input_buffer = ''
-        while re.search(prompt, input_buffer) is None:
-            input_buffer += self.shell.recv(16384)
+        start_time = time.time()
+        received_all_data = False
+
+        while not received_all_data:
+            recv_ready_timeout = (time.time() - start_time) >= self.ssh_cmd_timeout
+            # if pipe is empty and not timeout yet, keep waiting
+            should_wait = (not recv_ready_timeout) and (not self.shell.recv_ready())
+            if should_wait:
+                time.sleep(0.5)
+                continue
+
+            try:
+                input_buffer += self.shell.recv(16384)
+            except Exception as err:
+                msg = 'Receive ssh command result error: cmd={} msg={} type={}'.format(cmd, err, type(err))
+                self.log(msg)
+                return msg
```

Comment:
Why return the error message from `do_cmd` as `cmd` execution output in case of exception? Is it better to return `""`?

Reply:
I thought of it, but I think "" just like the 0 return from shell,
seems the command succeeded but actually not,
An error msg maybe helpful to do some exception control outside, even though there is not currently.
If someday I change it to catch the exception from outside for some need, I will change it from returning msg to raising exception.